src/motion/utils/apply-exit-props.ts

```diff
@@ -0,0 +1,26 @@
+import { MotionProps } from "../types"
+import { ExitProps } from "../context/MotionContext"
+import { invariant } from "hey-listen"
+
+export function applyExitProps<P>(
+    props: P & MotionProps,
+    { initial, isExiting, custom, onExitComplete }: ExitProps
+): P & MotionProps {
+    if (isExiting) {
+        invariant(!!props.exit, "No exit animation defined.")
+
+        return {
+            ...props,
+            custom: custom !== undefined ? custom : props.custom,
+            animate: props.exit,
+            onAnimationComplete: () => {
```

Comment:
This could potentially override user props right?

Reply:
Yeah it does so by design - I'll add a comment